MeshLib/Elements/Element.cpp

```diff
@@ -239,4 +244,106 @@ bool isPointInElementXY(MathLib::Point3d const& p, Element const& e)
          MeshLib::MeshElemType2String(e.getGeomType()).c_str());
     return false;
 }
+
+std::vector<std::size_t> findElementsInRadius(Element const& start_element,
+                                              double const radius)
+{
+    // Special case for 0 radius. All other radii will include at least one
+    // neighbor of the start element.
+    if (radius == 0.)
+        return {start_element.getID()};
+
+    // Returns true if the test node is inside the radius of any of the
+    // element's nodes.
+    auto node_inside_radius = [&start_element, &radius](Node const* test_node) {
+        for (int n = 0; n < start_element.getNumberOfNodes(); ++n)
+        {
+            if (MathLib::sqrDist(*test_node, *start_element.getNode(n)) <=
+                radius * radius)
+                return true;
+        }
+        return false;
+    };
+
+    std::set<std::size_t> found_elements;
```

Comment:
Can't that be a std::vector? Because visited_elements should provide uniqueness already.